### PR TITLE
feat(discover): Do not use tags[...] syntax for tags

### PR DESCRIPTION
**Pull request by lynnagara**
Every requested column not found in SENTRY_SNUBA_MAP is now assumed to be a
tag column in Snuba

### CODE REVIEW

File: src/sentry/utils/snuba.py [hunk 1]
```diff
@@ -29,6 +29,18 @@
...
+    'project_id': 'project_id',
+    'message': 'message',
+    'issue': 'issue',
+    'group_id': 'group_id',
+    'timestamp': 'timestamp',
+    'time': 'time',
+    'deleted': 'deleted',
+    'retention_days': 'retention_days',
+    'type': 'type',
+    'version': 'version',
```
Comment (macqueen) [inline src/sentry/utils/snuba.py]: i added a couple of these here: https://github.com/getsentry/sentry/pull/10503, which i just merged

also, is issue actually a column? i don't have it locally, but could also be that my snuba is outdated

**Review by macqueen — COMMENTED** [review]
looks like a couple tests are failing

File: src/sentry/utils/snuba.py (hunk 1 above)
Comment (alex-hofsteede) [reply src/sentry/utils/snuba.py]: `issue` is a computed column, not actually in the clickhouse table. There is a little bit about it in the [readme](https://github.com/getsentry/snuba/blob/master/README.md)

File: src/sentry/utils/snuba.py [hunk 2]
```diff
@@ -29,6 +29,18 @@
...
 SENTRY_SNUBA_MAP = {
+    # general
+    'event_id': 'event_id',
+    'project_id': 'project_id',
+    'message': 'message',
+    'issue': 'issue',
+    'group_id': 'group_id',
+    'timestamp': 'timestamp',
+    'time': 'time',
+    'deleted': 'deleted',
```
Comment (alex-hofsteede) [inline src/sentry/utils/snuba.py]: This column should never be used by clients. All snuba queries will filter out deleted rows internally anyway

Comment (alex-hofsteede) [reply src/sentry/utils/snuba.py]: Same with
retention_days: used internally to figure out when to delete old data
group_id: clients should always use `issue` and not query `group_id` directly.
received: This is also for internal use, I don't think we have ever exposed this to customers?

Comment (alex-hofsteede) [reply src/sentry/utils/snuba.py]: I'm not sure that this is the place to restrict that, I guess removing it from this list would just mean that queries for `deleted` would become `tags[deleted]` which is not what we want either.

File: src/sentry/static/sentry/app/views/organizationDiscover/data.jsx [hunk 3]
```diff
@@ -6,29 +6,29 @@ const TYPES = {
...
+  {name: 'os', type: TYPES.STRING},
+  {name: 'os_name', type: TYPES.STRING},
+  {name: 'os_rooted', type: TYPES.BOOLEAN},
+  {name: 'sentry:release', type: TYPES.STRING},
 ];
 
 // All tags are assumed to be strings, except the following
 export const SPECIAL_TAGS = {
-  'tags[os_rooted]': TYPES.BOOLEAN,
+  os_rooted: TYPES.BOOLEAN,
```
Comment (alex-hofsteede) [inline src/sentry/static/sentry/app/views/organizationDiscover/data.jsx]: If `PROMOTED_TAGS` contains the type information, I don't think we need a separate map here enumerating the ones that aren't strings?

**Review by alex-hofsteede — CHANGES_REQUESTED** [review]

File: src/sentry/static/sentry/app/views/organizationDiscover/data.jsx (hunk 3 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/organizationDiscover/data.jsx]: Good point, removed all the types on the promoted tag list (it's only loaded when tags query fails)

File: src/sentry/utils/snuba.py (hunk 2 above)
Comment (lynnagara) [reply src/sentry/utils/snuba.py]: I don't think any of this is going to be exposed intentionally, but it's always possible for someone to call the API with this directly or via search box input in event stream. If the terms group_id, deleted, retention_days are never communicated or prompted in anyway, is it safe to assume the someone who has intentionally input these terms is probably looking for a custom tag with that name that they've created?

Comment (alex-hofsteede) [reply src/sentry/utils/snuba.py]: yeah that makes sense. I guess the best we can do is turn `"deleted"` into `tags[deleted]` to see if its a custom tag for that user.

Actually that makes me wonder. if someone creates a custom tag with the same name as one of our columns, is there any way for them to query for it in discover? IDK if that's a priority or anything, just wondering.

Comment (lynnagara) [reply src/sentry/utils/snuba.py]: We could add support for the old syntax if that's a priority

Comment (alex-hofsteede) [reply src/sentry/utils/snuba.py]: nah, I don't mind, was just wondering.

**Review by alex-hofsteede — APPROVED** [review]
lgtm

**Pull request merged**